designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Button.kt

```diff
@@ -0,0 +1,180 @@
+package org.hisp.dhis.mobileui.designsystem.component
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.ripple.LocalRippleTheme
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ElevatedButton
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import org.hisp.dhis.mobileui.designsystem.theme.Outline
+import org.hisp.dhis.mobileui.designsystem.theme.Radius
+import org.hisp.dhis.mobileui.designsystem.theme.Ripple
+import org.hisp.dhis.mobileui.designsystem.theme.Spacing
+import org.hisp.dhis.mobileui.designsystem.theme.SurfaceColor
+import org.hisp.dhis.mobileui.designsystem.theme.TextColor
+import org.hisp.dhis.mobileui.designsystem.theme.buttonShadow
+
+/**
+ * DHIS2 button with generic icon slot.
+ * @param enabled Controls the enabled state of the button. When `false`, this button will not be
+ * clickable and will appear disabled to accessibility services.
+ * @param style Controls style of the button. Will be OUTLINED by default, but can be
+ * FILLED, TEXT, TONAL, ELEVATED or KEYBOARDKEY
+ * @param text The text to display within.
+ * @param icon The button icon content.
+ * @param onClick Will be called when the user clicks the button.
+ */
+@Composable
+fun Button(
+    enabled: Boolean = true,
+    style: ButtonStyle = ButtonStyle.OUTLINED,
+    text: String,
+    icon: @Composable
+    (() -> Unit)? = null,
+    onClick: () -> Unit
+) {
+    val paddingValues = getPaddingValues(icon != null)
+
+    when (style) {
+        ButtonStyle.FILLED -> {
+            val textColor = if (enabled) TextColor.OnPrimary else TextColor.OnDisabledSurface
+
+            Button(
```

Comment:
I realised that FILLED, TEXT, TONAL and KEYBOARDKEY use the same component (Button) changing the colors and the modifier. Do you think that we could extract this usage of the Button component to a private function and send the colors and modifiers as parameter?

Reply:
I have extracted it to private function SimpleButton